src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -368,9 +368,10 @@ CATCH_RETURN()
 try
 {
     const auto shift = gsl::narrow_cast<u8>(_api.lineRendition != LineRendition::SingleWidth);
-    const auto y = gsl::narrow_cast<u16>(clamp<til::CoordType>(coordTarget.y, 0, _p.s->cellCount.y));
-    const auto from = gsl::narrow_cast<u16>(clamp<til::CoordType>(coordTarget.x << shift, 0, _p.s->cellCount.x - 1));
-    const auto to = gsl::narrow_cast<u16>(clamp<size_t>((coordTarget.x + cchLine) << shift, from, _p.s->cellCount.x));
+    const auto x = std::max(0, coordTarget.x - (_p.s->viewportOffset.x >> shift));
+    const auto y = gsl::narrow_cast<u16>(clamp<til::CoordType>(coordTarget.y, 0, _p.s->viewportCellCount.y));
+    const auto from = gsl::narrow_cast<u16>(clamp<til::CoordType>(x << shift, 0, _p.s->viewportCellCount.x - 1));
+    const auto to = gsl::narrow_cast<u16>(clamp<size_t>((x + cchLine) << shift, from, _p.s->viewportCellCount.x));

```

Comment:
Here, the only change is that `coordTarget.x` is replaced with `x`.
`coordTarget.x` is in textbuffer-relative coordinates (so when you scroll horizontally, the left edge of the viewport might be something non-0), but we need viewport relative coordinates where the left edge is always 0.
To do so we first need to horizontal scroll-offset `viewportOffset.x` from the `coordTarget.x`, but this breaks with line renditions, because the caller already divides `coordTarget.x` by 2, whereas `viewportOffset.x` is not being divided by 2.

And so we first shift `viewportOffset.x` to the right, then subtract and then finally shift it to the left again to get viewport-relative, unshifted coordinates.